tools/castle-editor/code/formproject.pas

```diff
@@ -453,6 +459,46 @@ procedure TProjectForm.ListOutputClick(Sender: TObject);
   // TODO: just to source code line in case of error message here
 end;
 
+procedure TProjectForm.UpdateUndo(Sender: TObject);
+begin
+  UpdateUndoRedoInformation;
+end;
+
+procedure TProjectForm.UpdateUndoRedoInformation;
+begin
+  if (Design <> nil) and Design.UndoSystem.IsUndoPossible then
+  begin
+    MenuItemUndo.Enabled := true;
+    MenuItemUndo.Caption := Design.UndoSystem.UndoComment;
+  end else
+  begin
+    MenuItemUndo.Enabled := false;
+    MenuItemUndo.Caption := 'Undo is not possible';
```

Comment:
In this case, simpler caption seems more standard (checked with GIMP now).

So just `MenuItemUndo.Caption := 'Undo';`

The fact that it's grayed out already conveys the meaning of "it is not possible to click this" :)

Same for `'Redo is not possible'` below, should be just `'Redo'`.

Reply:
Done!